src/OutputWriters/jld2_output_writer.jl

```diff
@@ -31,7 +31,7 @@ ext(::Type{JLD2OutputWriter}) = ".jld2"
                               dir = ".",
                           indices = (:, :, :),
                        with_halos = false,
-                       array_type = Array{Float64},
+                             type = Float32,
```

Comment:
```suggestion
                             float_type = Float32,
```

a little more verbose, if we do go this direction. I think `array_type` is more explicitly regarding what actually happens in the code though, and provides more flexibility (I don't know what the application is, but I'm also not omniscient!)

Reply:
What about both `array_type = Array` and `float_type = Float32`?

Reply:
ah, that will work provided `convert(array_type{float_type}, output)` works I think.

Reply:
That looks all possible
```julia
julia> struct Test
           array_type
           float_type
       end

julia> T = Test(Array,Float32)
Test(Array, Float32)

julia> T.array_type{T.float_type}(undef,2,2)
2×2 Matrix{Float32}:
 0.0  0.0
 0.0  0.0

julia> A = rand(2,2)
2×2 Matrix{Float64}:
 0.203629  0.73571
 0.576096  0.28416

julia> convert(T.array_type{T.float_type}, A)
2×2 Matrix{Float32}:
 0.203629  0.73571
 0.576096  0.28416
```